### PR TITLE
Added ID duplication check function

### PR DESCRIPTION
**Pull request by royeom**
New functions are below:

**ID duplication check**  : In `register.py`, Check that the username entered as input already exists as a key in the DB.  If it doesn't exist send informations to DB.

Please take a look.

### CODE REVIEW

File: research/backend/restful/register.py [hunk 1]
```diff
@@ -29,11 +29,15 @@ def post(self):
         else:
             hashed = bcrypt.hashpw(password, bcrypt.gensalt())
 
-        # Insert in RedisDB
-        dbmanager.REDIS_DB.hmset( 'username:' + username, {
```
Comment (harrydrippin) [inline research/backend/restful/register.py:33]: Stick your first parameter and opening parenthesis together.

File: research/backend/restful/register.py [hunk 2]
```diff
@@ -29,11 +29,15 @@ def post(self):
...
-            'password': hashed, 
-            'firstname': first_name,
-            'lastname': last_name
+        # Check if input username already exists in DB
+        if dbmanager.REDIS_DB.keys('username:' + username):
+            raise ValueError('Your username is already being used.')
+        else:
+            # if not exist, insert in RedisDB
+            dbmanager.REDIS_DB.hmset( 'username:' + username, {
```
Comment (harrydrippin) [inline research/backend/restful/register.py]: Same as above.

**Review by harrydrippin — COMMENTED** [review]
Good! Take a look.

File: research/backend/restful/register.py (hunk 1 above)
Comment (royeom) [reply research/backend/restful/register.py:33]: I fixed it, thanks!

File: research/backend/restful/register.py (hunk 2 above)
Comment (royeom) [reply research/backend/restful/register.py]: fixed it, too!

**Review by harrydrippin — APPROVED** [review]
LGTM! 😄

**Pull request merged**